### PR TITLE
Query: Fixes NotImplementedException in OrderByCrossPartitionQueryPipelineStage

### PR DESCRIPTION
**Pull request by neildsh**
## Description
Currently we throw a NotImplementedException in OrderByCrossPartitionQueryPipelineStage if the continuation token contains any of the extended types like CosmosBinary and CosmosGuid. This visitor that throws was added so that we correctly handle continuation from undefined values. This change will cause us to use a coarse grained IS_DEFINED filter if we encounter these extended types instead of throwing. Currently, the extended types are only used by Cassandra which is schematized, so this should not have a performance penalty. We can look into adding more system functions like IS_BINARY etc. if we decide to expose these types through the SQL api.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs [hunk 1]
```diff
@@ -1175,6 +1175,12 @@ private static class IsSystemFunctions
                 public const string Object = "IS_OBJECT";
             }
 
+            private static class ExtendedTypesIsSystemFunctions
+            {
+                public const string Undefined = "not IS_DEFINED";
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs:1180]: >not [](http://example.com/codeflow?start=49&length=3)

NOT
(also above)

File: Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs [hunk 2]
```diff
@@ -1175,6 +1175,12 @@ private static class IsSystemFunctions
                 public const string Object = "IS_OBJECT";
             }
 
+            private static class ExtendedTypesIsSystemFunctions
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs:1178]: >ExtendedTypesIsSystemFunctions [](http://example.com/codeflow?start=33&length=30)

I don't think we should redefine those;; rather, we should reference IsSystemFunctions strings

**Review by sboshra — APPROVED** [review]
:shipit:

File: Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs (hunk 2 above)
Comment (neildsh) [reply Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs:1178]: The `IsSystemFunctions` strings are more fine grained and don't include `IS_DEFINED`, so I created the new one to stay consistent with the style in this class

Comment (neildsh) [reply Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CrossPartition/OrderBy/OrderByCrossPartitionQueryPipelineStage.cs:1178]: will make the change

**Pull request merged**